File: 2735-collecting-chocolates/2735-collecting-chocolates.java
class Solution {
    public long minCost(int[] nums, int x) {
        int n = nums.length;
        int[] A = new int[n];
        for(int i = 0; i < n; i += 1) A[i] = Integer.MAX_VALUE;
        long ans = Long.MAX_VALUE;
        for(int i = 0; i < n; i += 1) {
            
            long s = 0;
            for(int j = 0; j < n; j += 1) {
                A[j] = Math.min(A[j], nums[(j + i) % n]);
                s += A[j];
            }
            s += (long)i * (long)x;
            ans = Math.min(ans,s);
        }
        return ans;
    }
}

File: 2735-collecting-chocolates/2735-collecting-chocolates.cpp
class Solution {
public:
    long long minCost(vector<int>& v, int x) {
        int n = v.size();
        vector<int> A(n,INT_MAX);
        long long ans = LONG_MAX;
        for(int i = 0; i < n; i += 1) {
            long long s = 0;
            for(int j = 0; j < n; j += 1) {
                A[j] = min(A[j],v[(i + j) % n]);
                s += A[j];
            }
            s += (long long)i * (long long)x;
            ans = min(ans,s);
        }
        return ans;
    }
};